### PR TITLE
Allow manipulating namespaced indices on postgresql with the migration DSL

### PR DESCRIPTION
**Pull request by toommz**
### Motivation / Background

This Pull Request has been created because we use namespaced indices in our database, resulting in indices looking like this:

`namespace.index_table_on_column`

Trying to use the ActiveRecord migration DSL to remove such an index fails because the `Utils.extract_schema_qualified_name` method returns only `namespace` and not `namespace.index_table_on_column`.

### Detail

This Pull Request changes `Utils.extract_schema_qualified_name` to handle this case.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] There are no typos in commit messages and comments.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Feature branch is up-to-date with `main` (if not - rebase it).
* [x] Pull request only contains one commit for bug fixes and small features. If it's a larger feature, multiple commits are permitted but must be descriptive.
* [x] Tests are added if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.
* [x] PR is not in a draft state.
* [x] CI is passing.

### CODE REVIEW

**Comment by yahonda** [comment]
Hi, Thank you for opening this pull request.

As far as I understand, the term used here "namespace" is not official PostgreSQL term. Using the quoted identifier, the object name can include the dot.

https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
> SQL identifiers and key words must begin with a letter (a-z, but also letters with diacritical marks and non-Latin letters) or an underscore (_). 

https://www.postgresql.org/docs/current/sql-syntax-lexical.html
> quoted identifier. It is formed by enclosing an arbitrary sequence of characters in double-quotes ("). 

Is there any reason not to use the PostgreSQL standard schema as a namespace?

**Comment by toommz** [comment]
> Is there any reason not to use the PostgreSQL standard schema as a namespace?
> 
> 

I'm not sure to fully understand your question but in this case we use two schemas, `public` and `ml`.

Both are migrated from the same rails application. Without this little change it is impossible to act on indexes contained in the `ml` schema with the rails migration DSL.

File: activerecord/test/cases/adapters/postgresql/utils_test.rb [hunk 1]
```diff
@@ -16,7 +16,8 @@ def test_extract_schema_qualified_name
       %(schema."table_name")   => %w{schema table_name},
       %("schema"."table_name") => %w{schema table_name},
       %("even spaces".table)   => ["even spaces", "table"],
-      %(schema."table.name")   => ["schema", "table.name"]
+      %(schema."table.name")   => ["schema", "table.name"],
```
Comment (matthewd) [inline activerecord/test/cases/adapters/postgresql/utils_test.rb:19]: ☝🏻 this is already supported with the above syntax; does it not work for your use case?

Comment (toommz) [reply activerecord/test/cases/adapters/postgresql/utils_test.rb:19]: Hmmm not sure. Given an index created like this:

```ruby
add_index 'ml.matching_suggestions', :column, name: 'index_ml.matching_suggestions_unicity'
```

I can't remove it with the following migration:

```ruby
# Intended because the index name is not standard
remove_index 'ml.matching_suggestions', %i[document_id suggested_document_id]
```

Nor with:

```ruby
# This is where the PR comes to help
remove_index 'ml.matching_suggestions', name: 'index_ml.matching_suggestions_unicity'
```

I know it's an edge case and we can just pass raw SQL. This PR is now old and if you don't see  the added value of this I can close it without being emotional 😉

Comment (matthewd) [reply activerecord/test/cases/adapters/postgresql/utils_test.rb:19]: Ah okay, it's definitely true that `add_index` and `remove_index` should agree on the meaning of their arguments.

I think I would expect both to want `'"index_ml.matching_suggestions_unicity"'` here. 🤔
